qiskit/synthesis/evolution/suzuki_trotter.py

```diff
@@ -61,12 +64,21 @@ def __init__(
             reps: The number of time steps.
             insert_barriers: Whether to insert barriers between the atomic evolutions.
             cx_structure: How to arrange the CX gates for the Pauli evolutions, can be "chain",
-                where next neighbor connections are used, or "fountain", where all qubits are connected
-                to one.
+                where next neighbor connections are used, or "fountain", where all qubits are
+                connected to one.
             atomic_evolution: A function to construct the circuit for the evolution of single
                 Pauli string. Per default, a single Pauli evolution is decomopsed in a CX chain
                 and a single qubit Z rotation.
         """
+        if order % 2 == 1:
+            warnings.warn(
+                "SuzukiTrotter for odd orders is deprecated as of 0.20.0, and will be "
```

Comment:
What is the intent after the deprecation period - presumably instead of doing a warning here this if test will be kept and result instead in an error being raised about it requiring to be even? If so maybe put the commented out exception in here with a note that this is to replace the warning otherwise I am afraid that if we are not careful this all will just get removed after the deprecation period. Maybe at that time we should have a unit test that for an odd order that is expected to fail.

Reply:
Yes that's the intention, I've added the error commented out in the following line.